src/NCronJob/Configuration/NCronJobOptionBuilder.cs

```diff
@@ -154,51 +162,183 @@ internal static string GenerateJobName(Delegate jobDelegate)
+        return new NotificationStage<TJobDefinition>(services, settings, jobs, jobOptionBuilder);
+    }
+
+    /// <inheritdoc />
+    public INotificationStage<TJob> AddNotificationHandler<TJobNotificationHandler>() where TJobNotificationHandler : class, IJobNotificationHandler<TJob>
+    {
+        services.TryAddScoped<IJobNotificationHandler<TJob>, TJobNotificationHandler>();
+        return new NotificationStage<TJob>(services, settings, jobs, jobOptionBuilder);
+    }
+
+    /// <inheritdoc />
+    public IStartupStage<TNewJob> AddJob<TNewJob>(Action<JobOptionBuilder>? options = null) where TNewJob : class, IJob => new NCronJobOptionBuilder(services, settings, jobs).AddJob<TNewJob>(options);
+}
+
+/// <summary>
+/// Represents a stage in the job lifecycle where notifications are handled for the job.
+/// </summary>
+/// <typeparam name="TJob">The type of the job for which notifications are handled.</typeparam>
+public class NotificationStage<TJob> : INotificationStage<TJob> where TJob : class, IJob
+{
+    private readonly IServiceCollection services;
+    private readonly ConcurrencySettings settings;
+    private readonly List<JobDefinition> jobs;
+    private readonly JobOptionBuilder jobOptionBuilder;
+
+    internal NotificationStage(IServiceCollection services, ConcurrencySettings settings, List<JobDefinition> jobs, JobOptionBuilder jobOptionBuilder)
+    {
+        this.services = services;
+        this.settings = settings;
+        this.jobs = jobs;
+        this.jobOptionBuilder = jobOptionBuilder;
+    }
+
+    /// <inheritdoc />
+#pragma warning disable S1133 // Used to warn users not our internal usage
+    [Obsolete("The job type can be automatically inferred. Use AddNotificationHandler<TJobNotificationHandler> instead.", error: false)]
+#pragma warning restore S1133
+    public INotificationStage<TJobDefinition> AddNotificationHandler<TJobNotificationHandler, TJobDefinition>()
+        where TJobNotificationHandler : class, IJobNotificationHandler<TJobDefinition>
+        where TJobDefinition : class, IJob
+    {
+        services.TryAddScoped<IJobNotificationHandler<TJobDefinition>, TJobNotificationHandler>();
+        return new NotificationStage<TJobDefinition>(services, settings, jobs, jobOptionBuilder);
+    }
+
+    /// <inheritdoc />
+    public INotificationStage<TJob> AddNotificationHandler<TJobNotificationHandler>() where TJobNotificationHandler : class
+        , IJobNotificationHandler<TJob>
     {
+        services.TryAddScoped<IJobNotificationHandler<TJob>, TJobNotificationHandler>();
+        return this;
     }
 
+    /// <inheritdoc />
+    public IStartupStage<TNewJob> AddJob<TNewJob>(Action<JobOptionBuilder>? options = null) where TNewJob : class, IJob =>
+        new NCronJobOptionBuilder(services, settings, jobs).AddJob<TNewJob>(options);
+}
+
+/// <summary>
+/// Defines the contract for a stage in the job lifecycle.
+/// </summary>
+public interface IJobStage
```

Comment:
As this is never returned from any builder-method, does it need this interface in the first place?

Reply:
Ahh now I see.